test/services/helpers/service.test.js

```diff
@@ -1,93 +1,71 @@
 	const app = await appPromise;
 	const mailService = app.service('/mails');
+	let nestMailService;
 
-	afterEach(() => {
-		nock.cleanAll();
+	beforeEach(() => {
+		nestMailService = { send: sinon.spy() };
+		app.services['nest-mail'] = nestMailService;
 	});
 
 	describe('valid emails', () => {
 		it('should send an valid text email to the notification service', async () => {
-			const cb = getNotificationMock();
-			await mailService.create({
+			const params = {
 				email: 'test@test.test',
 				subject: 'test',
 				content: { text: 'Testing Purposes' },
-			});
+			};
+			await mailService.create(params);
 			// assert that notification service was actually called
-			expect(await cb).to.equal(true);
+			expect(nestMailService.send.calledOnce).to.eql(true);
+			const payload = nestMailService.send.firstCall.args[0];
+			expect(payload.recipients).to.eql([params.email]);
+			expect(payload.mail.plainTextContent).to.eql(params.content.text);
 		});
 		it('should send an valid html email to the notification service', async () => {
-			const cb = getNotificationMock();
-			await mailService.create({
+			const params = {
 				email: 'test@test.test',
 				subject: 'test',
 				content: { html: '<h1>Testing Purposes</h1>' },
-			});
+			};
+			await mailService.create(params);
 			// assert that notification service was actually called
-			expect(await cb).to.equal(true);
+			expect(nestMailService.send.calledOnce).to.eql(true);
+			const payload = nestMailService.send.firstCall.args[0];
+			expect(payload.recipients).to.eql([params.email]);
+			expect(payload.mail.htmlContent).to.eql(params.content.html);
 		});
 		it('files should be base64 encoded', async () => {
-			const cb = getNotificationMock({
-				attachments: [
-					{
-						filename: 'test.gif',
-						content: 'R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7',
-					},
-				],
-			});
-			await mailService.create({
+			const base64Content = 'R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
+			const params = {
 				email: 'test@test.test',
 				subject: 'test',
 				content: { html: '<h1>Testing Purposes</h1>' },
 				attachments: [
 					{
 						filename: 'test.gif',
-						content: Buffer.from('R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7', 'base64'),
+						content: Buffer.from(base64Content, 'base64'),
 					},
 				],
-			});
+			};
+			await mailService.create(params);
 			// assert that notification service was actually called
-			expect(await cb).to.equal(true);
+			expect(nestMailService.send.calledOnce).to.eql(true);
+			const payload = nestMailService.send.firstCall.args[0];
+			expect(payload.recipients).to.eql([params.email]);
+			expect(payload.mail.htmlContent).to.eql(params.content.html);
+			expect(payload.mail.attachments[0].name).to.eql(params.attachments[0].filename);
+			expect(payload.mail.attachments[0].base64Content).to.eql(base64Content);
+			expect(payload.mail.attachments[0].contentDisposition).to.eql('ATTACHMENT');
```

Comment:
Are the eql ( equal less ) operators the right operators here?
Especially for strings?